libs/remote_storage/src/lib.rs

```diff
@@ -152,9 +160,11 @@ pub trait RemoteStorage: Send + Sync + 'static {
     async fn list_prefixes(
         &self,
         prefix: Option<&RemotePath>,
+        timeout: Duration,
```

Comment:
I don't think that for a remote storage API one needs per-call specific timeout settings.

I think the timeout should be a global setting that is part of the struct and that can be edited via a `set_timeout` function, that's maybe part of the trait.

Reply:
Agreed this was a waste putting it on every operation. I'll just make that configurable via config.

Reply:
There is now a general configuration parameter for remote storage which is copied into per adapter configuration. There is no mutability.